lib/MatchingAlgorithm.cc

```diff
@@ -639,7 +621,7 @@ void CriticalRSM::ties_propose(FreeListType& free_list, VertexPtr a, PreferenceL
     }
 
     if (a_data.level == t && a_data.star) {
-      if (y < t || (y == t && (preference == 1 || preference == 2)) || (y == t && star && preference == 2)) {
+      if (y < t || (y == t && (preference == cEqual|| preference == cBetter)) || (y == t && star && preference == cBetter)) {
```

Comment:
Wrap complex statements in round braces, and consider defining booleans to simplify the if.
For e.g.
bool isBetter = (preference == cBetter);
bool isBetterOrEqual = (isBetter || (preference == cEqual));